plugins/emc/emc_vnx_block_lun_perfdata

```diff
@@ -0,0 +1,505 @@
+ env.blockpw foobar
+
+ Where: 
+ user - SSH Client local user
+ env.username - Remote user with Operator role for Block or File part
+ env.cs_addr - Control Stations addresses for remote (indirect) access.
+ env.localcli - Optional. Path of localhost 'Naviseccli' binary. If this 
+   variable is set, env.cs_addr is ignored, and local 'navicli' is used.
+   Requires env.blockpw variable.
+ env.sp_addr - Default is "SPA SPB". In case of "direct" connection to 
+   Storage Processors, their addresses/hostnames are written here.
+ env.blockpw - Password for connecting to Storage Processors
+
+=head1 ERRATA
+
+ It counts Queue Length in not fully correct way. We take parameters totally 
+ from both SP's, but after we divide them independently by load of SPA and SPB.
+ Anyway, in most AAA / ALUA cases the formula is correct.
+
+=head1 HISTORY
+
+ 09.11.2016 - First Release
+ 26.12.2016 - Compatibility with Munin coding style
+
+=cut
+
+export LANG=C
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+TARGET=$(echo "${0##*/}" | cut -d _ -f 6)
+
+# "All Storage Processors we have"
+if [[ -v "sp_addr" ]]; then
+	SPALL=$sp_addr
+else
+	SPALL="SPA SPB"
+fi
+# "navicli" command. Can be local or remote, through Control Stations
+if [[ -v "localcli" ]]; then
+	NAVICLI=$localcli
+else
+	NAVICLI="/nas/sbin/navicli"
+fi
+
+
+ssh_check_cmd() {
+		ssh -q $username@$1 "/nasmcd/sbin/getreason | grep -w slot_\`/nasmcd/sbin/t2slot\` | cut -d- -f1"
+}
+
+
+check_conf_and_set_vars () {
+	if [ -z "$username" ]; then
+		echo "No username ('username' environment variable)!"
+		return 1
+	fi
+
+	if [ -z "$localcli" ]; then
+		if [ -z "$cs_addr" ]; then
+			echo "No control station addresses ('cs_addr' environment variable)!"
+			return 1
+		fi
+		#Choosing Cotrol Station. Code have to be "10"
+		for CS in $cs_addr; do
+			if [[ "10" -eq "$(ssh_check_cmd $CS)" ]]; then
+				PRIMARY_CS=$CS
+				SSH="ssh -q $username@$PRIMARY_CS "
+				break
+			fi
+		done
+
+		if [ -z "$PRIMARY_CS" ]; then
+			echo "No alive primary Control Station from list \"$cs_addr\"";
+			return 1
+		fi
+	else
+		if [ ! -f "$localcli" ]; then
+			echo "Local CLI is set, but no binary found at $localcli!"
+			return 1
+		fi
+		if [ -z "$blockpw" ]; then
+			echo "No Password for Block Access ('blockpw' environment variable)!"
+			return 1
+		fi
+		SSH=""
+		NAVICLI="$localcli -User $username -Password $blockpw -Scope 0 "
+	fi
+	local probe_sp
+	for probe_sp in $SPALL; do
+		if $SSH $NAVICLI -h $probe_sp >/dev/null 2>&1; then
+			StorageProcessor=$probe_sp
+			break
+		fi
+	done
+	[ -z "$StorageProcessor" ] && echo "No active Storage Processor found!" && return 1
+	NAVICLI="$NAVICLI -h $StorageProcessor"
+	return 0
+}
+
+if [ "$1" = "autoconf" ]; then
+	check_conf_ans=$(check_conf_and_set_vars)
+		if [ $? -eq 0 ]; then
+			echo "yes"
+		else
+			echo "no ($check_conf_ans)"
+		fi
+		exit 0
+fi
+
+check_conf_and_set_vars 1>&2 || exit 1
+
+run_remote() {
+	if [ -z "$SSH" ]; then
+		eval "$@"
```

Comment:
But the problem is not with the first `$BIGCMD`, it's about the second command `getcontrol -cbt ...`, where due to the ugliness of navicli we have to call it separately for each SP. But SP addressess are "unknown" for this part of code (well, they are known since the array `$SPALL` is global, but it's not nice to call it here), so we have to call `..../\${SP}_total_busyticks.value....`, where `$SP` is backslashed. And the only way to evaluate it, which is known for me, is to use `eval`. If you see another way without rewriting the whole logic, please tell me.